db/view/view.cc

```diff
@@ -1109,102 +1124,65 @@ future<> mutate_MV(
                 return make_ready_future<>();
             }
         };
-        if (paired_endpoint) {
-            // If paired endpoint is present, remove it from the list of pending endpoints to avoid duplicates
-            pending_endpoints.erase(std::remove(pending_endpoints.begin(), pending_endpoints.end(), *paired_endpoint), pending_endpoints.end());
-            // When paired endpoint is the local node, we can just apply
-            // the mutation locally, unless there are pending endpoints, in
-            // which case we want to do an ordinary write so the view mutation
-            // is sent to them as well.
-            auto my_address = utils::fb_utilities::get_broadcast_address();
-            bool is_endpoint_local = *paired_endpoint == my_address;
-            int64_t updates_pushed_remote = !is_endpoint_local + pending_endpoints.size();
-
-            stats.view_updates_pushed_local += is_endpoint_local;
-            stats.view_updates_pushed_remote += updates_pushed_remote;
-            cf_stats.total_view_updates_pushed_local += is_endpoint_local;
-            cf_stats.total_view_updates_pushed_remote += updates_pushed_remote;
 
-            if (is_endpoint_local && pending_endpoints.empty()) {
-                // Note that we start here an asynchronous apply operation, and
-                // do not wait for it to complete.
-                // Note also that mutate_locally(mut) copies mut (in
-                // frozen form) so don't need to increase its lifetime.
-                // send_to_endpoint() below updates statistics on pending
-                // writes but mutate_locally() doesn't, so we need to do that here.
-                ++stats.writes;
-                auto mut_ptr = std::make_unique<frozen_mutation>(std::move(mut.fm));
-                tracing::trace(tr_state, "Locally applying view update for {}.{}; base token = {}; view token = {}",
-                        mut.s->ks_name(), mut.s->cf_name(), base_token, view_token);
-                future<> local_view_update = service::get_local_storage_proxy().mutate_locally(mut.s, *mut_ptr, std::move(tr_state), db::commitlog::force_sync::no).then_wrapped(
-                        [&stats,
-                         maybe_account_failure = std::move(maybe_account_failure),
-                         mut_ptr = std::move(mut_ptr)] (future<>&& f) {
-                    --stats.writes;
-                    return maybe_account_failure(std::move(f), utils::fb_utilities::get_broadcast_address(), true, 0);
-                });
-                fs->push_back(std::move(local_view_update));
+        // First, find the local endpoint and ensure that if it exists,
+        // it will be the target endpoint. That way, all endpoints in the
+        // pending_endpoints list are guaranteed to be remote.
```

Comment:
nitpick: outdated comment: the name "pending_endpoints" has changed in this version.